# Init

```{r}
suppressPackageStartupMessages({
  library(Seurat)
  library(Signac)
  library(tidyverse)
  library(chromVAR)
  library(JASPAR2020)
  library(TFBSTools)
  library(motifmatchr)
  library(BSgenome.Mmusculus.UCSC.mm39)
})

mbrain <- readRDS("data/mbrain_seurat_celltypes.rds")
```


```{r}
# Scan the DNA sequence of each peak for the presence of each motif, and create a Motif object
DefaultAssay(pbmc) <- "ATAC"
pwm_set <- getMatrixSet(x = JASPAR2020, opts = list(species = 10090, all_versions = FALSE))
motif.matrix <- CreateMotifMatrix(features = granges(mbrain), pwm = pwm_set, genome = 'mm39', use.counts = FALSE)
motif.object <- CreateMotifObject(data = motif.matrix, pwm = pwm_set)
mbrain <- SetAssayData(mbrain, assay = 'ATAC', slot = 'motifs', new.data = motif.object)

mbrain <- RunChromVAR(
  object = pbmc,
  genome = BSgenome.Mmusculus.UCSC.mm39
)
```



```{r}
markers_rna <- presto:::wilcoxauc(LayerData(mbrain, assay = "SCT", layer = "data"), mbrain$celltype)

markers_motifs <- presto:::wilcoxauc(LayerData(mbrain, assay = "ATAC", layer = "data"), group_by = 'celltype', assay = 'data', seurat_assay = 'chromvar')

motif.names <- markers_motifs$feature

colnames(markers_rna) <- paste0("RNA.", colnames(markers_rna))

colnames(markers_motifs) <- paste0("motif.", colnames(markers_motifs))

markers_rna$gene <- markers_rna$RNA.feature

markers_motifs$gene <- ConvertMotifID(pbmc, id = motif.names)
```


