# Init

```{r}
suppressPackageStartupMessages({
  library(Seurat)
  library(Signac)
  library(tidyverse)
  library(chromVAR)
  library(JASPAR2024)
  library(TFBSTools)
  library(motifmatchr)
  library(BSgenome.Mmusculus.UCSC.mm39)
})

# load Seurat object
mbrain <- readRDS("data/mbrain_seurat_celltypes.rds")
DefaultAssay(mbrain) <- "ATAC"

# load JASPAR motif set
sq24 <- RSQLite::dbConnect(RSQLite::SQLite(), db(JASPAR2024()))
pwm_set <- getMatrixSet(x = sq24, opts = list(species = 10090, all_versions = FALSE))
RSQLite::dbDisconnect(sq24)
```


## Motif Accessibility

A current set of mouse transcription factor motifs is downloaded from JASPAR and chromVAR is run to define per cell motif accessibility.

```{r}
motif.matrix <- CreateMotifMatrix(features = granges(mbrain), pwm = pwm_set, genome = 'mm39', use.counts = FALSE) # assign motifs to peaks from ATAC
motif.object <- CreateMotifObject(data = motif.matrix, pwm = pwm_set)
mbrain <- SetAssayData(mbrain, assay = 'ATAC', layer = 'motifs', new.data = motif.object) # add motifs object to the Seurat object

# run chromVAR on the motifs, i.e. find per cell motif accessibility scores
mbrain <- RunChromVAR(
  object = mbrain,
  genome = BSgenome.Mmusculus.UCSC.mm39
)
```

## get Transcription factor Expression and Motif markers for celltypes

```{r}
# get differentially expressed genes per cell type
markers_rna <- presto:::wilcoxauc(LayerData(mbrain, assay = "SCT", layer = "data"), mbrain$celltype)
colnames(markers_rna) <- paste0("RNA.", colnames(markers_rna))
markers_rna$gene <- markers_rna$RNA.feature

# get differentially accessible motifs per cell type
markers_motifs <- presto:::wilcoxauc(LayerData(mbrain, assay = "chromvar", layer = "data"), mbrain$celltype)
motif.names <- markers_motifs$feature
colnames(markers_motifs) <- paste0("motif.", colnames(markers_motifs))
markers_motifs$gene <- ConvertMotifID(mbrain, id = motif.names) # get gene names associated with motifs
```

## combine into marker dataframe

```{r}
celltypes <- levels(mbrain)

celltype.markers <- lapply(celltypes, function(celltype){
  ctmarkers_rna <- dplyr::filter(
    markers_rna, RNA.group == celltype, RNA.padj < 1e-2, RNA.logFC > 0) %>% 
    arrange(-RNA.auc)
  ctmarkers_motif <- dplyr::filter(
    markers_motifs, motif.group == celltype, motif.padj < 1e-2, motif.logFC > 0) %>% 
    arrange(-motif.auc)
  top_tfs <- inner_join(
    x = ctmarkers_rna[, c(2, 11, 6, 7)], 
    y = ctmarkers_motif[, c(2, 1, 11, 6, 7)], by = "gene"
  )
  top_tfs$avg_auc <- (top_tfs$RNA.auc + top_tfs$motif.auc) / 2
  top_tfs <- arrange(top_tfs, -avg_auc)
  return(top_tfs)
}) %>% bind_rows() %>% 
  mutate(celltype = RNA.group) %>% 
  select(-c(RNA.group, motif.group))

celltype.markers %>%
  ggplot(aes(x=celltype, y=avg_auc)) +
  geom_boxplot() +
  ylim(0.5, 1)
```

# Interesting genes

I pick out a couple of interesting TF genes based on their combined Gene Expression and motif accessibility to serve as markers for their assigned cell type.

## Helper functions

```{r}
plot.GEX.ATAC <- function(gene_name){
  motif.name <- ConvertMotifID(mbrain, name = gene_name)
  gene_plot <- FeaturePlot(mbrain, features = str_c("sct_", gene_name), reduction = 'wnn.umap')
  motif_plot <- FeaturePlot(mbrain, features = motif.name, min.cutoff = 0, cols = c("lightgrey", "darkred"), reduction = 'wnn.umap')
  gene_plot | motif_plot
}

plot.access <- function(gene_name, selection = "Astrocytes|Neuron|Oligode"){
  comp1 <- grep(selection, celltypes, value = TRUE)
  CoveragePlot(
    subset(mbrain, idents=comp1),
    region = gene_name, features = gene_name, assay = 'ATAC', expression.assay = 'SCT', peaks = FALSE, extend.upstream = 500, extend.downstream = 500)
}
```


## Dlx5

Dlx5 is a member of DLX gene family that encompasses multiple homeobox transcription factors. Mutations of it are associated with the split hand and foot malformation syndrome. It is known to be expressed in the brain, however a function specific to GABAergic neurons is not known.

```{r}
plot.GEX.ATAC("Dlx5")

plot.access("Dlx5")
```

## Plagl1

Plagl1 stands out with an average AUC of 0.85 for Astrocytes. It is associated with neural development in mice, but its association with astrocytes in particular seems new. Additionally, the ATAC profile shows its Transcrption Start Site (TSS) not being more accessible than other cell types, however it is distinguished by multiple peaks within exons. This might be a hint for this gene's expression being regulated on the RNA level by alternative splicing mechanisms.

```{r}
plot.GEX.ATAC("Plagl1")

plot.access("Plagl1")
```


## Neurod2

Neurod2 has been classified by my analysis as a marker gene for Neural Progenitor cells, but also has high expression/accessibility in cells on the way to become glutamatergic neurons. The gene is known for its role in neuron developement, but I could not find any information on it being more important for Glutamatergic than GABAergic developement.

```{r}
plot.GEX.ATAC("Neurod2")

plot.access("Neurod2", "Neuron|Progen")
```






