# Init

```{r}
suppressPackageStartupMessages({
  library(Seurat)
  library(Signac)
  library(tidyverse)
  library(GenomeInfoDb)
})
```

# Read feature file

```{r}
counts.10x <- Read10X_h5("../e18_mouse_multiOmic/outs/filtered_feature_bc_matrix.h5")

# extract RNA and ATAC data
rna_counts <- counts.10x$`Gene Expression`
atac_counts <- counts.10x$Peaks

print(paste0("There are ", ncol(rna_counts), " columns (cells) and ", nrow(rna_counts)," features (genes) in the Gene expression (scRNA-seq) data." ))
print(paste0("There are ", ncol(atac_counts), " columns (cells) and ", nrow(atac_counts)," features (peaks) in the Peaks (scATAC-seq) data." ))
```

# Create Seurat object

```{r}
# make Seurat object with rna data
mbrain <- CreateSeuratObject(counts = rna_counts)

# keep only canonical ('standard') chromosomes from the atac_counts data
atac_counts.grange <- StringToGRanges(rownames(atac_counts), sep = c(":", "-"))
tokeep <- as.vector(seqnames(atac_counts.grange) %in% standardChromosomes(atac_counts.grange))
atac_counts <- atac_counts[tokeep, ] 

# load annotation
gtf <- rtracklayer::import("../data/GRCm39/genes/genes.gtf.gz")
seqlevelsStyle(gtf) <- "UCSC"
genome(gtf) <- "mm39"
gtf$gene_biotype <- gtf$gene_type # fix name

# create Chromatin Assay
chrom.Assay <- CreateChromatinAssay(
  counts = atac_counts,
  sep = c(":", "-"),
  genome = "mm39",
  fragments = "../e18_mouse_multiOmic/outs/atac_fragments.tsv.gz",
  min.cells = 10,
  annotation = gtf
)

# add chromatin assay to the Seurat Object
mbrain[["ATAC"]] <- chrom.Assay
```

## QC plot

```{r}
VlnPlot(mbrain, features = c("nCount_ATAC", "nCount_RNA"), layer = "counts",  pt.size = 0, log=T)
```

## Subsetting based on QC

```{r}
# subset to only retain cells within the main count peak
mbrain <- subset(
  x = mbrain,
  subset = nCount_ATAC < 1e5 &
    nCount_ATAC > 3e3 &
    nCount_RNA < 5e4 &
    nCount_RNA > 1e3
)
```

# Normalisation and dimensionality reduction

## RNA

```{r}
# RNA
# normalize -> PCA -> UMAP
DefaultAssay(mbrain) <- "RNA"
mbrain <- SCTransform(mbrain) %>%
  RunPCA() %>%
  RunUMAP(dims = 1:50, reduction.name = 'umap.rna', reduction.key = 'rnaUMAP_')
```

## ATAC

```{r}
# ATAC analysis
DefaultAssay(mbrain) <- "ATAC"
mbrain <- mbrain %>% 
  RunTFIDF() %>% # TF-IDF normalisation
  FindTopFeatures(min.cutoff = 'q0') %>% # select all peaks
  RunSVD() %>% # dim reduction -> LSI space
  RunUMAP(reduction = 'lsi', dims = 2:50, reduction.name = "umap.atac", reduction.key = "atacUMAP_") # exclude first dimension for Umap
```

## weighted Nearest Neighbour (WNN) and clustering

```{r}
# WNN reduction and clustering of result
mbrain <- mbrain %>% 
  FindMultiModalNeighbors(reduction.list = list("pca", "lsi"), dims.list = list(1:50, 2:50)) %>% # use RNA and ATAC dim reduction for WNN
  RunUMAP(nn.name = "weighted.nn", reduction.name = "wnn.umap", reduction.key = "wnnUMAP_") %>% 
  FindClusters(graph.name = "wsnn", algorithm = 4, verbose = FALSE, random.seed = 1994) # Leiden algorithm for clustering
```

```{r}
Idents(mbrain) %>% 
  table()
```


## Plotting

```{r}
p1 <- DimPlot(mbrain, reduction = "umap.rna", label = TRUE, label.size = 2.5, repel = TRUE, pt.size = 1e-2) + ggtitle("RNA")
p2 <- DimPlot(mbrain, reduction = "umap.atac", label = TRUE, label.size = 2.5, repel = TRUE, pt.size = 1e-2) + ggtitle("ATAC")
p3 <- DimPlot(mbrain, reduction = "wnn.umap", label = TRUE, label.size = 2.5, repel = TRUE, pt.size = 1e-2) + ggtitle("WNN")
p1 + p2 + p3 & NoLegend() & theme(plot.title = element_text(hjust = 0.5))
ggsave("../figures/umap.compare.RNA.ATAC.WNN.png", dpi=300, width = 8, height = 4)
```


```{r}
# save seurat object for other scripts
saveRDS(mbrain, file = "../data/mbrain_seurat.rds")
```


# Session Info

```{r}
devtools::session_info()
```